### ISSUE TITLE
Doing lazy loading in tables

### ISSUE THREAD

**Issue by dve** [akquinet/vaadinator#31]
Right now we do a Service.findAll() to populate our tables. This is not a very good solution, because imagine a relay big result set and some concurrent users - you would keep your database in memory some times and that might be too much.
Also i would question if filtering and sorting should be done in the view level of the application or better in the backend.
What I'm not sure about is how to fit this in the MVP architecture?
- One way would be creating a container like the [LazyQueryContainer Add-on](https://github.com/tlaukkan/vaadin-lazyquerycontainer/issuesl), but this would totally ignore the presenter. 
- Adding onFilter-, onPageing and onSortingChange methods to the observer interface and calling them from the view by a bunch of listeners would respect the architecture, but would force a pageing table, not one which loads the data on demand on scrolling
- Implementing a container, which delegates loading, sorting and filtering to the presenter might be the cleaner approach but seems to be quite a lot of methods from the vaadin container interfaces to implement, so I would like to here about your thoughts on it first

**Comment by sebastianrothbucher** [comment]
One can in fact fit this into MVP by calling back the observer when new rows are needed. We didn't do it so far not to cause too much confusion at the beginning (who scrolls down so far). Need some thinking as per se, for larger use cases, you have to get it implemented. You totally have a point here..

**Comment by dve** [comment]
Can you take a look into https://github.com/dve/vaadinator/tree/lazyTable if this is a direction to go? Its not complete yet, but the earlier there is feedback the easier it is to change the direction.

**Comment by sebastianrothbucher** [comment]
Just checked your code and re-checked the writeup we did on this (https://blog-de.akquinet.de/2014/10/25/vaadin-unter-last-teil-4-performance-verbesserung-durch-lazy-loading/) and I think you're headed in the right direction. What I've experienced when doing it is that it's way more complex than one would initially think (though the LazyQueryContainer itself is designed quite well). Maybe it's worthwhile to fully implement w/out a generator first. In general though, asking the Observer (though you're following the pattern only 99%) is a way I'd consider valuable and OK...
